code/controllers/subsystem/aura.dm

```diff
@@ -90,15 +119,17 @@ SUBSYSTEM_DEF(aura)
 /datum/aura_bearer/process()
 	if(suppressed)
 		suppressed = FALSE
+		last_tick = world.time
 		return
 	if(affects_humans)
 		pulse_humans()
 	if(affects_xenos)
 		pulse_xenos()
 	if(duration < 0) //Negative duration means infinite. Usually for pheromones.
 		return
-	duration--
-	if(!duration)
+	duration -= (world.time - last_tick)
+	last_tick = world.time
+	if(duration <= 0)
```

Comment:
Is there a benefit to this over just calculating desired ending world.time at aura start and checking against it?

Reply:
suppression is supposed to freeze duration, no? That would be impossible to handle with that method which is why I chose this.

Reply:
Suppression interacting with duration wasn't actually something I put thought into when setting this up, since only xenos use suppression and their duration is always infinite. The purpose I had in mind was just to make the aura in question not emit for a tick.

Reply:
"Whether we should skip the next tick. Set to false after skipping once. Won't pulse to targets **or reduce duration**.
I simply preserved the functionality in the var's autodoc comment.